backend/src/agents/goal_achieved_agent.py

```diff
@@ -0,0 +1,10 @@
+from .agent import Agent, agent_metadata
+
+@agent_metadata(
+    name="GoalAchievedAgent",
+    prompt="",
+    description="Pick this agent if you believe you have solved BOTH the Current Task and the Next Task",
```

Comment:
Just a detail but the description for other agents is about what the agent does 'this agent does this' rather than an instruction to another agent, so to follow the same pattern we could have something like 'This agent informs the user that the goal has been achieved'

Reply:
It doesn't directly return to the user the result of the process, more that it acts as a choice for the router that, when chosen, has the side-affect that the process terminates. I do agree it can be named more in line with the other agents however.

Have changed to `"This agent triggers the end of solving the overall problem and returns the final answer to the user"`

Reply:
Sounds good and makes sense, thanks Charlie